libraries/opensk/src/api/crypto/ec_signing.rs

```diff
@@ -58,32 +58,16 @@ pub trait SecretKey: Sized {
     fn import(bytes: &[u8]) -> Option<Self>;
 }
 
-/// ECDSA verifying key.
+/// Elliptic curve verifying key.
 pub trait PublicKey: Sized {
     type Signature: Signature;
 
-    /// Creates a public key from its coordinates.
-    fn from_coordinates(x: &[u8; EC_FIELD_SIZE], y: &[u8; EC_FIELD_SIZE]) -> Option<Self>;
-
-    /// Verifies if the signature matches the message.
-    ///
-    /// For hashing, SHA256 is used implicitly.
-    fn verify(&self, message: &[u8], signature: &Self::Signature) -> bool;
-
-    /// Verifies if the signature matches the hash of the message.
-    ///
-    /// Prehash is the SHA256 of the signed message.
-    fn verify_prehash(&self, prehash: &[u8; HASH_SIZE], signature: &Self::Signature) -> bool;
-
     /// Writes the public key coordinates into the passed in parameters.
     fn to_coordinates(&self, x: &mut [u8; EC_FIELD_SIZE], y: &mut [u8; EC_FIELD_SIZE]);
```

Comment:
This is not correct for Ed25519. A public key is only 32 bytes.

Reply:
The only thing we do with public keys in the library is convert them to COSE. In `CoseKey`, we filled y with 0 in the case of Ed25519. This does not conform to the COSE RFC, and the field for y is now an `Option`.

That being said, I found other problems with our implementation that made the stack not work with Ed25519 only. Can be tested on [webauthn.io](https://webauthn.io/) for example using Advanced Settings.